src/ApiCheck/Program.cs

```diff
@@ -0,0 +1,254 @@
+                if (string.Equals(allowedBreakingChanges.Value(), "Binary", StringComparison.OrdinalIgnoreCase))
+                {
+                    // App will keep compiling after update, but apps compiled against the
+                    // old version of the assembly will have to be recompiled to work properly
+                    // after the update. I.E: Adding a parameter to an existing method with a default value.
+                    breakingChangeTypes = BreakingChangeTypes.Binary;
+                }
+                else if (string.Equals(allowedBreakingChanges.Value(), "Source", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Existing apps compiled against the old version of the assembly will continue to work
+                    // with the updated version. However, recompiling existing apps against the new assembly
+                    // will require changes on the source code. I.E: Introducing a method overload that creates
+                    // ambiguities to the app using it.
+                    breakingChangeTypes = BreakingChangeTypes.Source;
+                }
+                else if (string.Equals(allowedBreakingChanges.Value(), "All", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Fail whenever there is a breaking change of any type.
+                    breakingChangeTypes = BreakingChangeTypes.All;
+                }
+                else if (!string.IsNullOrEmpty(allowedBreakingChanges.Value()))
+                {
+                    command.ShowHelp();
+                    return Error;
+                }
+            }
+
+            var logger = CreateLogger();
+
+            var resolvedPackagesFolder = packagesFolder.Value() ??
+                $"{Environment.ExpandEnvironmentVariables("%userprofile%")}/.nuget/packages";
+
+            var resolvedFramework = framework.Value() == "netcoreapp1.0" ?
+                FrameworkConstants.CommonFrameworks.NetCoreApp10 :
+                FrameworkConstants.CommonFrameworks.Net452;
+
+            var assembly = AssemblyLoader.LoadAssembly(
+                assemblyPath.Value(),
+                lockFile.Value(),
+                resolvedFramework,
+                resolvedPackagesFolder);
+
+            var generationFilters = new List<Func<TypeInfo, bool>>();
+            var reloadingFilters = new List<Func<TypeBaseline, bool>>();
+            if (publicOnly.HasValue())
+            {
+                generationFilters.Add(t => t.IsPublic || t.IsNestedPublic || t.IsNestedFamily);
+                reloadingFilters.Add(t =>
+                    t.Visibility == BaselineVisibility.Public ||
+                    t.Visibility == BaselineVisibility.Protected ||
+                    t.Visibility == BaselineVisibility.ProtectedInternal);
+            }
+
+            if (excludeInternalNamespace.HasValue())
+            {
+                generationFilters.Add(t => !t.Namespace.EndsWith("Internal"));
```

Comment:
Actually even that isn't exactly right: If there is an "Internal" segment anywhere in the namespace it's "internal", so the check isn't EndsWith, rather, it's `.Split(".").Any(ns => ns == "Internal")` or something like that.